app/decorators/models/solidus_subscriptions/spree/variant/auto_delete_from_subscriptions.rb

```diff
@@ -0,0 +1,32 @@
+# frozen_string_literal: true
+
+module SolidusSubscriptions
+  module Spree
+    module Variant
+      module AutoDeleteFromSubscriptions
+        def self.prepended(base)
+          base.after_discard(:remove_from_active_subscriptions)
+          base.after_destroy(:remove_from_active_subscriptions)
+        end
+
+        def remove_from_active_subscriptions
+          SolidusSubscriptions::LineItem.delete(related_active_subscriptions_line_item_ids)
+        end
+
+        def related_active_subscriptions_line_item_ids
+          query = <<-SQL
+            SELECT line_items.id
+            FROM #{SolidusSubscriptions::LineItem.table_name} AS line_items
+            JOIN #{SolidusSubscriptions::Subscription.table_name} AS subscriptions
+            ON line_items.subscription_id = subscriptions.id
+            WHERE line_items.subscribable_id = #{id}
+          SQL
+
+          ActiveRecord::Base.connection.exec_query(query).map { |row| row['id'] }
+        end
```

Comment:
Is there a reason you haven't simply done it like this?
```ruby
def remove_from_active_subscriptions
  SolidusSubscriptions::LineItem.where(subscribable: self).delete_all
end
```

Reply:
The query was a bit more complex at first and it seemed reasonable, right now it really isn't so I simplified it as you suggested, thanks 🙏